### PR TITLE
Fix and improve ServiceDispatchHandler#removeServicePeer

### PR DESCRIPTION
**Pull request by jcorbin**
r @kriskowal @anson627

### CODE REVIEW

**Comment by jcorbin** [comment]
The first commit fixes the uncaught we saw in prod.

The second two fix the semantics to be (more) correct.

**Comment by kriskowal** [comment]
:shipit:

File: node/hyperbahn/service_proxy.js [hunk 1]
```diff
@@ -296,12 +296,31 @@ function refreshServicePeer(serviceName, hostPort) {
...
+    var subChanKeys = Object.keys(self.channel.subChannels);
+    for (var i = 0; i < subChanKeys; i++) {
+        var subChan = self.channel.subChannels[subChanKeys[i]];
+        if (subChan.peers.get(hostPort)) {
+            anyOtherSubChan = true;
+            break;
+        }
+    }
+
+    if (!anyOtherSubChan) {
```
Comment (anson627) [inline node/hyperbahn/service_proxy.js]: so if there is any other sub channel, we don't do anything, meaning traffic will still go through. my original intention is to stop forwarding once we de-advertise.

Comment (jcorbin) [inline node/hyperbahn/service_proxy.js]: good point... guess what: the original never did that either, upcoming PR to fix... if only there was a unit test for this module ;-)

**Pull request merged**